lib/ocaml_kafka.c

```diff
@@ -317,22 +317,18 @@ value ocaml_kafka_new_topic(value caml_partitioner_callback, value caml_kafka_ha
      RAISE(rd_errno, "Failed to create new kafka topic (%s)", rd_kafka_err2str(rd_errno));
   }
 
-  // The handler is wrapped with its name.
   caml_kafka_topic_handler = alloc_caml_handler(topic);
-  caml_topic = caml_alloc_small(2,0);
-  Field(caml_topic, 0) = caml_kafka_topic_handler;
-  Field(caml_topic, 1) = caml_topic_name;
-  CAMLreturn(caml_topic);
+  CAMLreturn(caml_kafka_topic_handler);
 }
 
 extern CAMLprim
 value ocaml_kafka_destroy_topic(value caml_kafka_topic)
 {
   CAMLparam1(caml_kafka_topic);
 
-  rd_kafka_topic_t *topic = handler_val(Field(caml_kafka_topic,0));
+  rd_kafka_topic_t *topic = handler_val(caml_kafka_topic);
   if (topic) {
-    free_caml_handler(Field(caml_kafka_topic,0));
+    free_caml_handler(Field(caml_kafka_topic, 0));
```

Comment:
This must be replaced by `free_caml_handler(caml_kafka_topic);`

Reply:
Thanks, fixed it! Coming from OCaml to C it is really annoying that you can do whatever wrong thing with very little support from the compiler and refactoring things becomes dangerous.